src/robot/parsing/lexer/statementlexers.py

```diff
@@ -334,3 +334,17 @@ class BreakLexer(TypeAndArguments):
     @classmethod
     def handles(cls, statement: list, ctx: TestOrKeywordContext):
         return statement[0].value == 'BREAK'
+
+
+class SyntaxErrorLexer(TypeAndArguments):
+    token_type = Token.ERROR
+
+    @classmethod
+    def handles(cls, statement: list, ctx: TestOrKeywordContext):
+        return statement[0].value in {'END', 'ELSE', 'ELSE IF','EXCEPT'}
```

Comment:
As mentioned above, `BREAK`, `CONTINUE` and possibly also `RETURN` should be added here.